libraries/AP_Airspeed/AP_Airspeed_UAVCAN.cpp

```diff
@@ -95,11 +95,24 @@ AP_Airspeed_UAVCAN* AP_Airspeed_UAVCAN::get_uavcan_backend(AP_UAVCAN* ap_uavcan,
     }
 
     if (!detected) {
+
```

Comment:
I think we need an additional check that we haven't yet run the probe() function. Otherwise if a new airspeed sensor turns up after we've probed then we would be changing it's index. 
Simplest is to have a static bool that is true when probe() has been called and return early, not adding to the array if this is set

Reply:
Agree, added. (Need some handling for arming check now to ensure all the configured sensors were found at boot, but will make a separate pull request)